equinox/_module.py

```diff
@@ -478,11 +480,44 @@ def __new__(
                                     "attempting to override "
                                     f"`{base.__module__}.{base.__qualname__}.{k}`."
                                 )
+
         # [Step 6] Register as a pytree.
+        data_fs = tuple(
+            f.name
+            for f in dataclasses.fields(cls)
+            if not f.metadata.get("static", False)
+        )
+        static_fs = tuple(
+            f.name for f in dataclasses.fields(cls) if f.metadata.get("static", False)
+        )
+
+        def flatten_module(
+            obj: "Module", /, with_keys: bool
+        ) -> tuple[tuple[PyTree, ...], _FlattenedData]:
+            # Static metadata, placed in aux.
+            static_vs = [getattr(obj, name) for name in static_fs]
+            # Subnodes in the PyTree (filtering on `__dict__` to avoid picking
+            # up `property` and uninitialized values).
+            dynamic_fs = [name for name in data_fs if name in obj.__dict__]
+            if with_keys:
+                dynamic_vs = [(jtu.GetAttrKey(k), getattr(obj, k)) for k in dynamic_fs]
+            else:
+                dynamic_vs = [getattr(obj, k) for k in dynamic_fs]
```

Comment:
Since speed is a concern, note that this is doing two lookups into `obj.__dict__`, once in `name in obj.__dict__` and another in `getattr`. I think the former could be skipped by replacing the latter with some kind of `obj.__dict__.get(k, _some_sentinel_value)`.

(I also speculate that a small speedup could be arranged by assigning to a local variable:
```python
get = obj.__dict__.get
dynamic_vs = [get(k, _some_kind_of_sentinel) for k in dynamic_fs]
```
)

Reply:
Done!

Reply:
https://github.com/patrick-kidger/equinox/pull/994/commits/53e829297c55c675cc49b3472a091219d141fc52 only uses a single loop for getting from the dict.